text/0328-polyglot-assert.md

```diff
@@ -0,0 +1,336 @@
+---
+rfc pr: https://github.com/aws/aws-cdk-rfcs/pull/330
+tracking issue: https://github.com/aws/aws-cdk-rfcs/issues/328
+---
+
+# Polyglot Assert
+
+The `@aws-cdk/assert` module is only available to javascript users.
+
+The new polyglot assert module - `@aws-cdk/assertions` - provides the same set of
+functionalities to users in all CDK supported languages.
+
+## Working Backwards
+
+### CHANGELOG
+
+**feat:** announcing assertions: the assert library is now available in all CDK supported languages.
+
+### README
+
+<!--BEGIN STABILITY BANNER-->
+
+---
+
+![cdk-constructs: Stable](https://img.shields.io/badge/cdk--constructs-stable-success.svg?style=for-the-badge)
+
+---
+
+<!--END STABILITY BANNER-->
+
+> NOTE: This module contains *beta APIs*.
+>
+> All beta symbols are suffixed with the `Beta<n>`. When we have backwards
+> incompatible change, we will create a new symbol with a `Beta<n+1>` suffix
+> and deprecate the `Beta<n>` symbol.
+---
+This module allows asserting the contents of CloudFormation templates.
+
+To run assertions based on a CDK `Stack`, start off with -
+
+```ts
+const stack = new cdk.Stack(...)
+...
+const assert = TemplateAssertionsBeta1.fromStack(stack);
+```
+
+Alternatively, assertions can be run on an existing CloudFormation template -
+
+```ts
+const template = fs.readFileSync('/path/to/template/file');
+const assert = TemplateAssertionsBeta1.fromTemplate(template);
+```
+
+#### Full Template Match
+
+The simplest assertion would be to assert that the template matches a given
+template.
+
+```ts
+assert.assertMatchTemplateBeta1({
```

Comment:
My vote is in line with what Elad is saying - I wouldn't have the `Beta` preview suffix on the methods (having it one the class is enough).

Also, I would suggest changing the method names to read a little more fluently.

```suggestion
assert.assertTemplateMatches({
```

Reply:
Nice! I like the suggestion around fluency.